### ISSUE TITLE
Mint 21.1: Steam fails to start some games and throws GLi-GObject-CRITICAL every time

### ISSUE THREAD

**Issue by csarp2** [ValveSoftware/steam-for-linux#9718]
#### Your system information
Hardware: Ryzen 7 5800X, 32GB @ 3200MHz, Radeon RX 6600 XT (amd driver)
* Steam client version (build number or date): 1687386907
* Distribution (e.g. Ubuntu): Linux Mint (Ubuntu) 21.1
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes
* Steam Logs: [logs.zip](https://github.com/ValveSoftware/steam-for-linux/files/11857895/logs.zip)
* GPU: AMD

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

What should happen:
Before new UI update, I could open all games in my library with default Steam Play settings and they would run fine.

What does happen:
After this update, about half of my (small) library fails to start, namely No Man's Sky, Among Us and ROUNDS. Others (like Just Shapes & Beats, Super Auto Pets and Celeste) run fine. When run from terminal, the critical error is always the same regardless of the game:

`(process:5378): GLib-GObject-CRITICAL **: 15:38:46.133: g_object_ref: assertion 'G_IS_OBJECT (object)' failed`
 
`(process:5378): GLib-GObject-CRITICAL **: 15:38:46.133: g_object_unref: assertion 'G_IS_OBJECT (object)' failed`

The following has been tested and has not fixed the issue:
- Clearing Download Cache
- Reinstalling Steam and renaming the .steam folder so as to generate a new one
- Reinstalling the games
- For No Man's Sky, deleting the whole BINARIES folder to force-generate new files
- Verifying the integrity of each game
- Forcing the use of Proton 6, 7, 8 and experimental - Proton 6 gave [no error messages](https://gist.github.com/csarp2/34e7f50e68d420b000b0d2de357e51b9) but still did not launch
- Repairing the download directory
- Resetting all steam settings to defaults
- adding launch arg `PROTON_USE_WINED3D=1 %command%`

Some terminal outputs from things I've tried (I've run all non-functional games under each conditions and the outputs are all basically the same, so I'm just putting No Man's Sky outputs here for simplicity):
[Not forcing proton version, no launch arguments](https://gist.github.com/csarp2/c025befcf21d56b99c9a5b9fbeaf88c6)
[Not forcing proton version, PROTON_USE_WINED3D=1 %command% in launch arguments](https://gist.github.com/csarp2/0e4fca3975c5e878e5c7fa82dbf8d8b3)

I can run all games on my Windows laptop.
#### Steps for reproducing this issue:

1. Install Linux Mint 21.1 and Steam
2. Enable Steam Play for all titles
3. Download No Man's Sky, Among Us or ROUNDS
4. Attempt to run the game through terminal as root using `steam steam://rungameid/#######`

**Comment by oneoa** [comment]
Hello! Might be related to #9605

**Comment by Gamebuster19901** [comment]
> Hello! Might be related to #9605

No… Those were marked off topic because those comments were for a completely different issue.

#9321 is what those comments were being directed to. Whether or not it’s actually the same issue, I don’t know.
